doc/holochain_101/src/planning_a_dapp.md

```diff
@@ -1 +1,68 @@
+
+## What is a dApp? 
+A dApp is a distributed application. This means that the data associated with the application is stored by each user rather than in a central database.  
+
+## Basic expectations for dApps
+Because app data storage is distributed amongst user-participants, one must expect that data encryption and permissions are important for protecting privacy in accordance with the jurisdictions in which the app is operating.
+
+Remember that, as user-participants leave the application, they take their data with them. They also retain copies of other data that they held to support the DHT.  
+
+One must also re-think the dApp's business model such that it does not rely on a central authority's ability to whitelist access to a given resource. 
+
+## Cryptography in holochain dApps
+Distributed systems rely more heavily on cryptography than centralized systems. 
+
+### Hashes
+
+### Signatures 
+
+### Encryption
+
+## Entry types 
+You can define particular entry types as public, private, or encrypted. Encryption is an in-built option for holochain apps, meaning that you can automatically encrypt entry types by default.
+This is because holochain's use of DHTs means that simply running the application allows user-participants access to all entries on the network.
+
+### Public entries
+Public entries are published to the app's DHT with an appropriate redundancy factor. 
+
+### Encrypted entries
+Like public entries, encrypted entries are published to the app's DHT with an appropriate redundancy factor.
+
+### Private entries
+Unlike public and encrypted entries, private entries are only on the source chain (device) of the publisher and are not shared out to the DHT. 
+Of course, one could always encrypt a private entry, too. 
+Consider the use cases of a private blockchain...
+
+## Security - best practices
+Different applications will require different levels of security. Holochain is uniquely suited to accommodate a high degree of flexibility, however there are generic best practices based on its native architecture. In essence, one must consider how to prevent undesired access to the DHT. After all, if someone has access to the source code, the app's DNA, they also have access to the DHT. This means that everyone who uses the app potentially has access to all the network's entries. Developers must treat the code as if it's a key to the data, effectively shifting their security efforts to the code itself rather than where sensitive data is stored. Though holochains rely on the cryptography above to create trust in data's provenance and immutability, trust is a distinctly human affair at the level of access to the code. A great way to begin offsetting the governance crises now typical of distributed systems (i.e. DAO hack) is to think in terms of protecting and enabling the community of user-participants in addition to cryptography.    
```

Comment:
That is not correct actually, I think.

A really important concept is **membranes**. And it is important to be able to explain why Holochain works. So, not everybody who has access to the DNA automatically can join a network. The DNA could define any kind of membrane (closes list of participants, PoW, Proof of Service, social triangulation.. or just open). If the new node does not pass through this membrane, no existing node would talk to that guy. Potentially not even answer DHT requests.

Reply:
Interesting...I've changed things around some to include a greater focus on membranes. It could be nice to have the "scenarios to consider" section give light suggestions for different use cases (just to model some membrane thinking). What do y'all think?

Reply:
I believe membranes as well as the related topic of the immune system deserve both a full sub-section. The problem with that is, we have neither build nor specified these things yet... Something to put on our plate soon, @zippy?